### PR TITLE
Implementing Flask WTForm

### PR DESCRIPTION
**Pull request by corpsgeek**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by corpsgeek** [comment]
This pull request is linked to https://github.com/section-io/engineering-education/issues/1140

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by corpsgeek** [comment]
@lalith1403 How's the review process going?

File: articles/implementing-flask-wtform/index.md [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+The following are the two methods of handling forms in Flask:
+- Request Object
+- Flask- WTF Extension
+
+# Form Creation With HTML Forms
+The first step to working with forms in your Flask application is to create your forms before handling them. You can create your forms using basic HTML and CSS. 
+
+However, for development and production purpose. The use of frameworks like Bootstrap because form error handling in Flask is easier.
+
+In our cloned Flask application, the HTML template is rendered from the app/views.py file, displaying a Hello World text. Now, we would undergo some coding sprint here to structure this application properly.
```
Comment (lalith1403) [inline articles/implementing-flask-wtform/index.md:56]: Rephrase the second sentence. We would undergo.... -> We use the above files to code the application.

File: articles/implementing-flask-wtform/index.md [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+
+First, create a new folder named **template** within the **app**  folder. This folder will consist of all our HTML files. Proceed further to creating a new HTML file named `register.html`. 
+
+In our register.html folder, we initialize the content with a complete HTML code that links to Bootstrap CSS. By following this process, your `register.html` file content should be similar to mine [here](https://gist.github.com/corpsgeek/18cb010ff2d051c4da15897d08083a45)
```
Comment (lalith1403) [inline articles/implementing-flask-wtform/index.md:60]: In the `register.html` file, we initialize the content ...

Comment (lalith1403) [reply articles/implementing-flask-wtform/index.md:60]: Also, should we go to the gist link and copy the code? It is not clear.

File: articles/implementing-flask-wtform/index.md [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+
+With the register.html file setup with Bootstrap, you won’t be using any bootstrap classes for the moment.
```
Comment (lalith1403) [inline articles/implementing-flask-wtform/index.md:62]: `register.html`

File: articles/implementing-flask-wtform/index.md [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+In the implementation of our flask form, we need to install the Flask-WTForms first. 
+
+```bash
+$ pip install flask-wtf
+```
+Next, open the `forms.py` file created earlier. Here, we import the Flask form
+```python
+from flask_wtf import FlaskForm
+```
+The subsequent step is to create a contact form. To do this, we initialize a class that inherits the FlaskForm object. Your code should look similar to this.
```
Comment (lalith1403) [inline articles/implementing-flask-wtform/index.md:180]: `FlaskForm` object. Whenever you refer to filenames, variables, or classes, make sure to use the proper formatting. Change this throughout the article.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
The code works well. A few changes related to formatting and grammar, and we should be good to go.

**Comment by corpsgeek** [comment]
@lalith1403 I have resolved the changes in the "Resolved Requested Changes". But, I am not sure if I made a mistake with fetching the data from the master branch and it seems to commit a merge, can you please resolve this from your end if possible cause I tried to, but I don't know if everything is still in order. Thank You.

**Comment by lalith1403** [comment]
@hectorkambow I can't check if there are merge conflicts in this branch. Can you check if there are any issues with this PR?

**Comment by corpsgeek** [comment]
@hectorkambow Good morning, am i good to go with regards the merge conflict? for further review by @lalith1403

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by lalith1403** [comment]
@corpsgeek Include a hero image. Once done, we can move onto the final round of review. Before pushing the file, pull the changes made and only then perform the push operation. I made the same changes as made in the earlier review. If you have any doubts, contact me, and I will help you out.

**Comment by corpsgeek** [comment]
@lalith1403 I added the hero image. You can proceed to the final review. Thank you.

**Comment by hectorkambow** [comment]
@lalith1403 great work - please see the minor edits I made to the front matter. 

👍

**Pull request merged**